tools/src/publish-packages/tasks/publishCanary.ts

```diff
@@ -52,6 +64,49 @@ export const prepareCanaries = new Task<TaskArgs>(
   }
 );
 
+const publishCanaryProjectTemplates = new Task<TaskArgs>(
+  {
+    name: 'publishCanaryProjectTemplates',
+    dependsOn: [prepareCanaries],
+  },
+  async (parcels: Parcel[], options: CommandOptions) => {
+    await runWithSpinner(
+      'Updating and publishing project templates',
+      async (step) => {
+        const templates = await getAvailableProjectTemplatesAsync();
+        const bundledNativeModules = require(BUNDLED_NATIVE_MODULES_PATH);
+        const expoVersion = await sdkVersionAsync();
+        const dependenciesToUpdate = {
+          ...bundledNativeModules,
+          expo: `~${expoVersion}`,
+        };
+
+        for (const template of templates) {
+          step.start(`Updating and publishing ${green(template.name)}`);
+
+          const templateView = await getPackageViewAsync(template.name);
+
+          // Canary project template uses the same version as the `expo` package that it depends on.
+          const canaryVersion = findNextAvailableCanaryVersion(
+            expoVersion,
+            templateView?.versions ?? []
+          );
+
+          // Update versions of the dependencies based on `bundledNativeModules.json`.
+          await updateTemplateVersionsAsync(template.path, canaryVersion, dependenciesToUpdate);
+
+          // Publish the project template with a `canary` tag.
+          await publishPackageAsync(template.path, {
+            tagName: 'canary',
```

Comment:
can we also add the sdk branch tag? so for sdk 51 canary, `sdk-51`

Reply:
Sure, we can! But I noticed there is an issue with how we version the `expo` package in canary releases – it's released as `50.0.0-canary-*` instead of 51 😞 So, I would fix that first so we don't accidentally override `sdk-50` tag.

Reply:
:X yeah we should definitely bump that to 51!